### PR TITLE
YD-554 Improve entity loading

### PR DESCRIPTION
**Pull request by Bert-R**
Made a considerable improvement

I feel we need to do more, especially related to loading the user profile and to message loading.

For message loading, we still need 28 queries in this break down:

| Method | Query count |
|---|-------|
| canAccessPrivateData | 2 |
| getPrivateValidatedUserEntity | 1 |
| tryTransferDirectMessagesToAnonymousDestination | 2 |
| tryProcessUnprocessedMessages | 2 |
| getPrivateValidatedUser | 14 |
| getReceivedMessages | 7 |

The first candidate seems to be to work with a partially loaded user DTO (``getPrivateValidatedUser ``).

### CODE REVIEW

File: analysisservice/src/test/java/nu/yona/server/analysis/service/ActivityUpdateServiceTest.java [hunk 1]
```diff
@@ -91,6 +94,8 @@
 	@Mock
 	private ActivityCacheService mockAnalysisEngineCacheService;
 	@Mock
+	private GoalRepository mockGoalRepository;
```
Comment (aristotelos) [inline analysisservice/src/test/java/nu/yona/server/analysis/service/ActivityUpdateServiceTest.java]: The `ActivityUpdateService` doesn't have `GoalRepository` dependency. So why is this necessary?

File: analysisservice/src/test/java/nu/yona/server/analysis/service/AnalysisEngineServiceTest.java [hunk 2]
```diff
@@ -104,6 +107,8 @@
 	@Mock
 	private ActivityCacheService mockAnalysisEngineCacheService;
 	@Mock
+	private GoalRepository mockGoalRepository;
```
Comment (aristotelos) [inline analysisservice/src/test/java/nu/yona/server/analysis/service/AnalysisEngineServiceTest.java]: Same comment here, this is not a dependency of `AnalysisEngineService`...

File: appservice/src/intTest/groovy/nu/yona/server/BuddyValidationTest.groovy [hunk 3]
```diff
@@ -166,7 +166,7 @@ class BuddyValidationTest extends AbstractAppServiceIntegrationTest
 		User richard = addRichard()
 		User bob = addBob()
 		bob.emailAddress = "bob@dunn.com"
-		appService.sendBuddyConnectRequest(richard, bob)
+		assertResponseStatusCreated(appService.sendBuddyConnectRequest(richard, bob))
```
Comment (aristotelos) [inline appservice/src/intTest/groovy/nu/yona/server/BuddyValidationTest.groovy]: Isn't it more logical to place this assertion inside the `sendBuddyConnectRequest` method?

File: core/src/main/java/nu/yona/server/messaging/service/MessageService.java [hunk 4]
```diff
@@ -189,14 +197,24 @@ private void deleteMessages(Collection<Message> messages)
...
+	}
+
+	private MessageSource getAnonymousMessageSource(User user)
+	{
+		return getMessageSource(user.getAnonymousMessageSourceId());
+	}
+
+	private MessageSource getMessageSource(UUID id)
+	{
+		return MessageSource.getRepository().findOne(id);
```
Comment (aristotelos) [inline core/src/main/java/nu/yona/server/messaging/service/MessageService.java]: It is better to add the `MessageSourceRepository` as a dependency.

File: core/src/main/java/nu/yona/server/subscriptions/entities/UserAnonymized.java [hunk 5]
```diff
@@ -151,4 +155,13 @@ public void removeDeviceAnonymized(DeviceAnonymized deviceAnonimized)
 	{
 		return devicesAnonymized;
 	}
+
+	public void preloadGoals()
+	{
+		if (!isGoalPreloadDone)
+		{
+			Goal.getRepository().findByUserAnonymizedId(getId()); // Preload in one go
```
Comment (aristotelos) [inline core/src/main/java/nu/yona/server/subscriptions/entities/UserAnonymized.java]: Can't we inject the `GoalRepository` dependency as a parameter in the `preloadGoals` method and pass it from the `UserAnonymizedService`?

File: core/src/main/java/nu/yona/server/util/HibernateStatisticsService.java [hunk 6]
```diff
@@ -318,5 +341,74 @@ public long getTransactionCount()
...
 			return transactionCount;
 		}
+
+		public List<String> getSqlStatements()
+		{
+			return sqlStatements;
+		}
+	}
+
+	private static class LogAppender extends AppenderBase<ILoggingEvent>
```
Comment (aristotelos) [inline core/src/main/java/nu/yona/server/util/HibernateStatisticsService.java]: Shouldn't we call this `SqlStatementLogAppender`?

File: analysisservice/src/test/java/nu/yona/server/analysis/service/ActivityUpdateServiceTest.java (hunk 1 above)
Comment (Bert-R) [reply analysisservice/src/test/java/nu/yona/server/analysis/service/ActivityUpdateServiceTest.java]: The method ``UserAnonymized.preloadGoals`` uses it and that's being called from the ``ActivityUpdateService``.

File: analysisservice/src/test/java/nu/yona/server/analysis/service/AnalysisEngineServiceTest.java (hunk 2 above)
Comment (Bert-R) [reply analysisservice/src/test/java/nu/yona/server/analysis/service/AnalysisEngineServiceTest.java]: See reply above.

File: appservice/src/intTest/groovy/nu/yona/server/BuddyValidationTest.groovy (hunk 3 above)
Comment (Bert-R) [reply appservice/src/intTest/groovy/nu/yona/server/BuddyValidationTest.groovy]: :anguished: It's already part of it. The last parameter is a boolean named ``assertSuccess``, with ``true`` as default. I'll rever this change.

File: core/src/main/java/nu/yona/server/messaging/service/MessageService.java (hunk 4 above)
Comment (Bert-R) [reply core/src/main/java/nu/yona/server/messaging/service/MessageService.java]: Agreed.

File: core/src/main/java/nu/yona/server/util/HibernateStatisticsService.java (hunk 6 above)
Comment (Bert-R) [reply core/src/main/java/nu/yona/server/util/HibernateStatisticsService.java]: If it were a top-level class, I would agree, but given the local context I don't see added value in a longer name.
